src/AWS.Deploy.Orchestration/DeploymentCommands/BeanstalkEnvironmentDeploymentCommand.cs

```diff
@@ -3,19 +3,66 @@
 
 using System;
 using System.Collections.Generic;
-using System.Text;
+using System.Linq;
 using System.Threading.Tasks;
 using AWS.Deploy.Common;
 using AWS.Deploy.Orchestration.DisplayedResources;
 
 namespace AWS.Deploy.Orchestration.DeploymentCommands
 {
-    /// <summary>
-    /// This class is a Work In Progress.
-    /// </summary>
     public class BeanstalkEnvironmentDeploymentCommand : IDeploymentCommand
     {
-        public Task ExecuteAsync(Orchestrator orchestrator, CloudApplication cloudApplication, Recommendation recommendation) => throw new NotImplementedException();
-        public Task<List<DisplayedResourceItem>> GetDeploymentOutputsAsync(IDisplayedResourcesHandler displayedResourcesHandler, CloudApplication cloudApplication, Recommendation recommendation) => throw new NotImplementedException();
+        public async Task ExecuteAsync(Orchestrator orchestrator, CloudApplication cloudApplication, Recommendation recommendation)
+        {
+            if (orchestrator._interactiveService == null)
+                throw new InvalidOperationException($"{nameof(orchestrator._interactiveService)} is null as part of the orchestartor object");
+            if (orchestrator._awsResourceQueryer == null)
+                throw new InvalidOperationException($"{nameof(orchestrator._awsResourceQueryer)} is null as part of the orchestartor object");
+            if (orchestrator._elasticBeanstalkHandler == null)
+                throw new InvalidOperationException($"{nameof(orchestrator._elasticBeanstalkHandler)} is null as part of the orchestartor object");
+            if (orchestrator._s3Handler == null)
+                throw new InvalidOperationException($"{nameof(orchestrator._s3Handler)} is null as part of the orchestartor object");
+
+            var deploymentPackage = recommendation.DeploymentBundle.DotnetPublishZipPath;
+            var environmentName = cloudApplication.Name;
+            var applicationName = (await orchestrator._awsResourceQueryer.ListOfElasticBeanstalkEnvironments())
+                .Where(x => string.Equals(x.EnvironmentId, cloudApplication.UniqueIdentifier))
+                .FirstOrDefault()?
+                .ApplicationName;
+
+            if (string.IsNullOrEmpty(applicationName))
+            {
+                var message = $"Could not find any Elastic Beanstalk application that contains the following environment: {environmentName}";
+                throw new AWSResourceNotFoundException(DeployToolErrorCode.FailedToFindElasticBeanstalkApplication, message);
+            }
+
+            orchestrator._interactiveService.LogMessageLine($"Initiating deployment to {cloudApplication.DisplayName}...");
+
+            var versionLabel = $"v-{DateTime.Now.Ticks}";
```

Comment:
nit: could you remove the dash between the "v" and ticks? versions are usually v1, v2, v3 and not v-1, v-2, v-3